### PR TITLE
Accessibility: Remove unnecessary onClick from title element

### LINKED ISSUE #59695: A11y: Fix keyboard accessibility in `public/app/core/components/QueryOperationRow/QueryOperationRowHeader.tsx`

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- removes unnecessary `onClick` from the title `<div>`
- this is handled by the `IconButton` immediately next to it anyway

**Why do we need this feature?**

- better a11y across grafana

**Who is this feature for?**

- everyone! 🙌 

**Which issue(s) does this PR fix?**:

Fixes #59695 

**Special notes for your reviewer**:

### CODE REVIEW

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Hmm this does change the functionality, because the `titleElement` is used in some places like "Query Options"

Screenshot 2022-12-22 at 11 35 18

And this change means that that title is not clickable anymore. Was that the intended change? If so we should probably make it not blue anymore because it looks clickable.

Alternatively that title should be a button or anchor link to make its role clear

**Comment by ashharrison90** [comment]
@JoaoSilvaGrafana mmmm, actually i'm leaning more towards making the whole thing a button... (query options + icon)... lemme try that

**Comment by JoaoSilvaGrafana** [comment]
> ah, can't make the whole thing a button as you then end up with nested interactive elements in some places

But that specific element is the `headerElement`, while the one that has the click is the `titleElement` (like in query options), could you make the `IconButton` + `titleElement` the clickable area? That shouldn't affect the query title interactive element

**Review by eledobleefe — APPROVED** [review]
LGTM! 🎉

**Comment by ashharrison90** [comment]
@JoaoSilvaGrafana yeah we can, but then we have a couple of other issues:
- there's nothing stopping someone passing an interactive `titleElement` somewhere (yay for allowing **anything**!)
- the hover state of the button only spans part of the row header which looks super weird imo

**Comment by JoaoSilvaGrafana** [comment]
> there's nothing stopping someone passing an interactive titleElement somewhere (yay for allowing anything!)

😱 😱 

Ok fair, I do have to say I don't think I ever used the expand button for the places that this will affect (Query options, data options and transformations), so as a user I would see the fact that the title is no longer clickable as a bug, but I do agree that from an accessibility point of view this makes more sense...

**Comment by ashharrison90** [comment]
@JoaoSilvaGrafana yeah i'm not happy with either solution tbh... maybe we leave this til after the holidays and have a longer look at it. or if you wanna take a look in the mean time, feel free! 👍

File: public/app/core/components/QueryOperationRow/QueryOperationRowHeader.tsx [hunk 1]
```diff
@@ -44,10 +46,13 @@ export const QueryOperationRowHeader: React.FC<QueryOperationRowHeaderProps> = (
           type="button"
           aria-expanded={isContentVisible}
           aria-controls={id}
+          aria-labelledby={titleId}
         />
         {titleElement && (
-          <div className={styles.titleWrapper} onClick={onRowToggle} aria-label="Query operation row title">
-            <div className={cx(styles.title, disabled && styles.disabled)}>{titleElement}</div>
+          <div className={styles.titleWrapper} aria-label="Query operation row title">
```
Comment (joshhunt) [inline public/app/core/components/QueryOperationRow/QueryOperationRowHeader.tsx:52]: could you get rid of this aria-label while you're at it? 🙏🏻

**Review by joshhunt — APPROVED** [review]

**Comment by torkelo** [comment]
this to me makes this component a bit harder to use, needs much more precision to expand. Can't we keep the click on the title?

**Pull request merged**